src/som/primitives/integer_primitives.py

```diff
@@ -105,6 +105,8 @@ def _leftShift(ivkbl, frame, interpreter):
     l = left.get_embedded_integer()
     r = right.get_embedded_integer()
     try:
+        if not (l == 0 or 0 <= r < LONG_BIT):
```

Comment:
Why do you do you check for `l == 0` here?

Shouldn't any shift then be a noop?